compose/mpp/demo/src/uikitMain/kotlin/androidx/compose/mpp/demo/UIKitInteropExample.kt

```diff
@@ -206,3 +240,56 @@ private fun ComposeUITextField(value: String, onValueChange: (String) -> Unit, m
         }
     )
 }
+
+val UIKitReusableMapsExample = Screen.Example("UIKitReusableMapsExample") {
+    var allocations: Int by remember { mutableStateOf(0) }
+
+    Column(Modifier.fillMaxSize()) {
+        Text("Maps allocated: $allocations")
+        LazyColumn(Modifier.fillMaxSize()) {
+            items(100) {
+                UIKitView(
+                    factory = {
+                        allocations += 1
+                        val view = MKMapView().apply {
+                            tag = allocations.toLong()
+                        }
+
+                        view
+                    },
+                    modifier = Modifier.fillMaxWidth().height(200.dp),
+                    update = {
+                        println("MKMapView updated")
+                    },
+                    onReset = {
+                        println("${it.tag} onReset")
+                    },
+                    onRelease = {
+                        allocations -= 1
+                    },
+                    listener = object : UIKitInteropListener<MKMapView> {
+                        override fun onDidAppear(component: MKMapView) {
+                            println("${component.tag} onDidAppear frame: ${NSStringFromCGRect(component.frame)}, isAttached = ${component.window != null}")
+                        }
+
+                        override fun onDidDisappear(component: MKMapView) {
+                            println("${component.tag} onDidDisappear frame: ${NSStringFromCGRect(component.frame)}, isAttached = ${component.window != null}")
+                        }
+
+                        override fun onWillAppear(component: MKMapView) {
+                            println("${component.tag} onWillAppear frame: ${NSStringFromCGRect(component.frame)}, isAttached = ${component.window != null}")
+                        }
+
+                        override fun onWillDisappear(component: MKMapView) {
+                            println("${component.tag} onWillDisappear frame: ${NSStringFromCGRect(component.frame)}, isAttached = ${component.window != null}")
+                        }
+
+                        override fun onResize(component: MKMapView, size: CValue<CGSize>) {
+                            println("${component.tag} onResize frame: ${NSStringFromCGRect(component.frame)}")
+                        }
```

Comment:
Wouldn't this listener object be recreated on each recomposition?

Reply:
And the worst thing is that it won't be reapplied. Revisiting that part again.